GeeksCoreLibrary/Modules/Databases/Services/MySqlDocumentStorageService.cs

```diff
@@ -0,0 +1,113 @@
+using GeeksCoreLibrary.Core.Models;
+using GeeksCoreLibrary.Modules.Databases.Interfaces;
+using GeeksCoreLibrary.Modules.Databases.Models;
+
+namespace GeeksCoreLibrary.Modules.Databases.Services;
+
+/// <inheritdoc />
+public class MySqlDocumentStorageService : IDocumentStorageService, IScopedService
+{
+    private readonly IDocumentStoreConnection documentStorageConnection;
+
+    public MySqlDocumentStorageService(IDocumentStoreConnection documentStorageConnection)
+    {
+        this.documentStorageConnection = documentStorageConnection;
+    }
+
+    /// <inheritdoc />
+    public async Task CreateCollection(string prefix = "")
+    {
+        List<(string name, DocumentStoreIndexModel index)> indices = new()
+        {
+            ("idx_id", new DocumentStoreIndexModel
+            {
+                Fields =
+                {
+                    new DocumentStoreIndexFieldModel
+                    {
+                        Field = "id",
+                        Required = true,
+                        Type = "BIGINT UNSIGNED"
+                    }
+                }
+            }),
+            ("idx_title", new DocumentStoreIndexModel
+            {
+                Fields =
+                {
+                    new DocumentStoreIndexFieldModel
+                    {
+                        Field = "title",
+                        Required = true,
+                        Type = "VARCHAR(255)"
+                    }
+                }
+            }),
+            ("idx_changedOn", new DocumentStoreIndexModel
+            {
+                Fields =
+                {
+                    new DocumentStoreIndexFieldModel
+                    {
+                        Field = "changedOn",
+                        Required = true,
+                        Type = "DATETIME"
+                    }
+                }
+            })
+        };
+        await documentStorageConnection.CreateCollectionAsync($"{prefix}{WiserTableNames.WiserItemStore}", indices);
+    }
+
+    /// <inheritdoc />
+    public async Task<(WiserItemModel model, string documentId)> StoreItemAsync(WiserItemModel wiserItem, EntitySettingsModel entitySettings = null)
+    {
+        var prefix = entitySettings?.DedicatedTablePrefix ?? String.Empty;
+        
+        wiserItem.ChangedOn = DateTime.Now;
+        
+        var id = await documentStorageConnection.InsertOrUpdateDocumentAsync($"{prefix}{WiserTableNames.WiserItemStore}", wiserItem, wiserItem.Id);
+
+        return (wiserItem, id);
+    }
+    
+    /// <inheritdoc />
+    public async Task<ulong> EmptyItemCollectionAsync(EntitySettingsModel entitySettings = null)
+    {
+        var prefix = entitySettings?.DedicatedTablePrefix ?? String.Empty;
+        return await documentStorageConnection.RemoveDocumentsAsync($"{prefix}wiser_item_store");
+    }
+
+    /// <inheritdoc />
+    public async Task DeleteItem(string documentId, EntitySettingsModel entitySettings = null)
```

Comment:
Please update the name to `DeleteItemAsync` (although this will probably happend automatically if you refactor the name in the interface).